camp/camp-base/src/main/java/org/apache/brooklyn/camp/spi/collection/BasicResourceLookup.java

```diff
@@ -43,6 +47,11 @@ public synchronized void add(T item) {
         if (old!=null) {
             items.put(old.getId(), old);
             throw new IllegalStateException("Already contains item for "+item.getId()+": "+old+" (adding "+item+")");
```

Comment:
Why does this override the previous value and then throw the exception? I'd have thought we'd either just throw, or we'd override and just log.warn.